components/script/script_thread.rs

```diff
@@ -2982,23 +3261,21 @@ impl ScriptThread {
     pub fn handle_tick_all_animations_for_testing(id: PipelineId) {
         SCRIPT_THREAD_ROOT.with(|root| {
             let script_thread = unsafe { &*root.get().unwrap() };
-            script_thread
-                .handle_tick_all_animations(id, AnimationTickType::CSS_ANIMATIONS_AND_TRANSITIONS);
+            let document = match script_thread.documents.borrow().find_document(id) {
+                Some(document) => document,
+                None => return warn!("Animation tick for tests for closed pipeline {}.", id),
+            };
+            document.maybe_mark_animating_nodes_as_dirty();
         });
     }
 
     /// Handles when layout finishes all animation in one tick
-    fn handle_tick_all_animations(&self, id: PipelineId, tick_type: AnimationTickType) {
+    fn handle_tick_all_animations(&self, id: PipelineId) {
         let document = match self.documents.borrow().find_document(id) {
             Some(document) => document,
             None => return warn!("Message sent to closed pipeline {}.", id),
         };
-        if tick_type.contains(AnimationTickType::REQUEST_ANIMATION_FRAME) {
-            document.run_the_animation_frame_callbacks();
-        }
-        if tick_type.contains(AnimationTickType::CSS_ANIMATIONS_AND_TRANSITIONS) {
-            document.maybe_mark_animating_nodes_as_dirty();
-        }
+        document.tick_all_animations();
```

Comment:
This function can be removed entirely now and the code that calls it can simply call `document.tick_all_animations()` which will make it impossible to not find the document.